databases/week2/part4erdiagram1.sql

```diff
@@ -0,0 +1,48 @@
+/*CREATE TABLE `Staff` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,  
+  `email` varchar(255) NOT NULL,
+  `phone` varchar(255) NULL, 
+  `office_id` int(10) NOT NULL,
+  `department_id` int(10) NOT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+*/
+
+INSERT INTO Staff (id, name, email, phone, office_id, department_id)  
+VALUES -- (1, 'John', 'john@dk.com', '12345', 1, 1),
+		(2, 'Lisa', 'lisa@dk.com', '12346', 2, 2),
+        (3, 'Louise', 'louise@dk.com', '12347', 3, 3);
+
+/*
+CREATE TABLE `Office` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `staff_id` int(10) unsigned NOT NULL,  -- primary key and fk key must have unsigned
+  `location` varchar(255) NOT NULL, 
+  `phone` int(10) NOT NULL,
+  PRIMARY KEY (`id`),
+  CONSTRAINT `fk_class` FOREIGN KEY (`staff_id`) REFERENCES `staff` (`id`) ON DELETE CASCADE ON UPDATE CASCADE 
```

Comment:
Again here is an issue with the Table names being defined with the first letter uppercase, but in the foreign key constraints they are in lowercase.

Reply:
Also more importantly: You should think about your relations. Does it make sense to have a staff_id foreign key here. This means there can only be one staff member in each office, but staff members can be in many offices.